File: backend/Domain/Administration/Project.cs
using System.ComponentModel.DataAnnotations;
using Conversey.BL.Domain.Common;
using Conversey.BL.Domain.Ideation;
using Conversey.BL.Domain.Survey;

namespace Conversey.BL.Domain.Administration;

public class Project
{
    [Required]
    public Slug Id { get; set; }

    [StringLength(100)]
    public string Name { get; set; }
    
    [StringLength(4000)]
    public string Description { get; set; }

    [StringLength(2048)]
    public string ImageUrl { get; set; }
    public Status Status { get; set; }
    
    [Range(0, 150)]
    public int? MinAge { get; set; }
    
    [Range(0, 150)]
    public int? MaxAge { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public InteractionType InteractionForm { get; set; }
    
    [Range(1, 5)]
    public int NudgingStrength { get; set; } = 3;
    
    public IEnumerable<Topic> Topic { get; set; }

    public ProjectTheme Theme { get; set; }

    public IEnumerable<Question> Questions { get; set; }

    [Required]
    public Workspace Workspace { get; set; }

    public IEnumerable<Youth> Youth { get; set; }

    public IEnumerable<Idea> ProjectIdeas { get; set; }
}

public enum Status
{
    Draft,
    Active,
    Archived
}

public enum InteractionType
{
    Chat,
    VerticalScroll,
    UserDefined
}

File: backend/UI-MVC/Assets/models/project.ts
export const ProjectStatus = {
    Draft: 'Draft',
    Active: 'Active',
    Archived: 'Archived',
} as const

export type ProjectStatus = (typeof ProjectStatus)[keyof typeof ProjectStatus]

export const InteractionType = {
    Chat: 'Chat',
    VerticalScroll: 'Vertical_Scroll',
    UserDefined: 'UserDefined',
} as const

export type InteractionType = (typeof InteractionType)[keyof typeof InteractionType]

export interface ProjectTheme {
    primary: string
    secondary: string
    accent: string
    preset: string
    font: string
}

export interface ProjectTopic {
    id: number
    name: string
    context: string
    maxBroadSelectionLoads: number
}

export interface Project {
    id: string
    slug: string
    organizationSlug: string
    organizationName?: string
    organizationLogo?: string
    title: string
    description: string
    imageUrl: string
    status?: ProjectStatus
    startDate?: string
    endDate?: string
    interactionType?: InteractionType
    nudgingStrength: number
    topic?: ProjectTopic
    topics?: ProjectTopic[]
    theme?: ProjectTheme
}

/* will need changes: like organizationslug and projectslug are just edited version of the titles.
could be calculated instead of stored.

maybe zstart date and end date, and maybe a status (active, archived, etc.)

topic and style isnt needed yet but will be in the future to load every project according to admin settings



*/
